relayer/src/transfer.rs

```diff
@@ -71,6 +75,57 @@ impl FromStr for Amount {
     }
 }
 
+#[derive(Copy, Clone)]
+pub enum TransferTimeout {
+    Height(Height),
+    Timestamp(Timestamp),
+    HeightAndTimestamp(Height, Timestamp),
+}
+
+impl TransferTimeout {
+    pub fn new(
+        timeout_height_offset: u64,
+        timeout_duration: Duration,
+        destination_chain_status: &ChainStatus,
+    ) -> Result<Self, TransferError> {
+        let offset_is_zero = timeout_height_offset == 0;
+        let timeout_is_zero = timeout_duration == Duration::ZERO;
+
+        let timeout_height = destination_chain_status.height.add(timeout_height_offset);
+
+        let timeout_timestamp = (destination_chain_status.timestamp + timeout_duration)
```

Comment:
I think we're correctly using the `destination_chain_status.timestamp` (i.e. the receiving chain's timestamp) here. 👍